### ISSUE TITLE
Proposal: `IfAny`, `IsAny`, `IsUnknown`, etc

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/type-fest#129]
https://github.com/sindresorhus/type-fest/pull/127/files#r474306457

Thoughts?

**Comment by Milo123459** [comment]
@sindresorhus I think this would be very helpful.

**Comment by sindresorhus** [comment]
What would you use it for?

**Comment by Milo123459** [comment]
I have a new package I'm working on, which works with objects. This would help me manage things. 

If this doesn't get added, I don't really mind, it's already an amazing library. Thanks @sindresorhus 👍

**Comment by sindresorhus** [comment]
Alright. This is PR welcome.

The submitter should make sure to follow https://github.com/sindresorhus/type-fest/blob/master/.github/contributing.md closely.

**Comment by sindresorhus** [comment]
If anyone wants to work on this, see the initial attempt in https://github.com/sindresorhus/type-fest/pull/505

Instead of `IfAny`, we could potentially do what's done in https://github.com/sindresorhus/type-fest/pull/544 Feedback wanted on that way of handling it.

**Comment by tommy-mitchell** [comment]
I've started working on this, but I'm struggling to think of use cases for these beyond being helpers in other types.

I also started by making both an `IsAny` and an `IfAny` in `is-if-any.d.ts` (etc.), but I found that unional/type-plus#201 combined them like so[^1]:

```ts
type IsAny<T, TypeIfAny = true, TypeIfNotAny = false> = 0 extends 1 & T ? TypeIfAny : TypeIfNotAny;

IsAny<any> //=> true
IsAny<number> //=> false
IsAny<any, number, string> //=> number
```

I kind of like that `IsAny` and `IfAny` are more explicit in what they do, but the succinctness of this method is nice. Thoughts?

[^1]: Really, [it was `IsNever`](https://github.com/unional/type-plus/blob/main/ts/PrimitiveTypes.ts#L6), but it's the same difference.
